clients/ui/frontend/src/app/pages/modelCatalogSettings/components/ManageSourceForm.tsx

```diff
@@ -0,0 +1,134 @@
+} from '~/app/pages/modelCatalogSettings/useManageSourceData';
+import { FORM_LABELS, DESCRIPTIONS } from '~/app/pages/modelCatalogSettings/constants';
+import SourceDetailsSection from './SourceDetailsSection';
+import CredentialsSection from './CredentialsSection';
+import YamlSection from './YamlSection';
+import ModelVisibilitySection from './ModelVisibilitySection';
+import PreviewPanel from './PreviewPanel';
+import ManageSourceFormFooter from './ManageSourceFormFooter';
+
+type ManageSourceFormProps = {
+  existingData?: Partial<ManageSourceFormData>;
+  isEditMode: boolean;
+};
+
+const ManageSourceForm: React.FC<ManageSourceFormProps> = ({ existingData, isEditMode }) => {
+  const navigate = useNavigate();
+  const [formData, setData] = useManageSourceData(existingData);
+  const [isFiltersExpanded, setIsFiltersExpanded] = React.useState(false);
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
+
+  const isHuggingFaceMode = formData.sourceType === SourceType.HuggingFace;
+  const isFormComplete = isFormValid(formData);
+  const canPreview = isPreviewReady(formData);
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    setSubmitError(undefined);
+
+    try {
+      // TODO: Implement submit logic (will be part of API integration)
+      // navigate(catalogSettingsUrl());
+    } catch (error) {
+      setSubmitError(error instanceof Error ? error : new Error('Failed to save source'));
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const handlePreview = () => {
+    // TODO: Implement preview logic (will be part of API integration)
+  };
+
+  const handleCancel = () => {
+    navigate(catalogSettingsUrl());
+  };
+
+  return (
+    <>
+      <Grid hasGutter span={12}>
+        <GridItem span={7}>
+          <Form isWidthLimited>
+            <Stack hasGutter>
+              <StackItem>
+                <SourceDetailsSection formData={formData} setData={setData} />
+              </StackItem>
+
+              {isHuggingFaceMode && (
+                <StackItem>
+                  <CredentialsSection formData={formData} setData={setData} />
+                </StackItem>
+              )}
+
+              {!isHuggingFaceMode && (
+                <StackItem>
+                  <YamlSection formData={formData} setData={setData} />
+                </StackItem>
+              )}
+
+              <StackItem>
+                <ModelVisibilitySection
+                  formData={formData}
+                  isExpanded={isFiltersExpanded}
+                  onToggle={() => setIsFiltersExpanded((prev) => !prev)}
+                  setData={setData}
+                />
+              </StackItem>
+
+              <StackItem>
+                <FormSection>
+                  <FormGroup fieldId="enable-source">
+                    <Checkbox
+                      label={FORM_LABELS.ENABLE_SOURCE}
```

Comment:
The label for this checkbox should be bold, you can pass in JSX to `label` for that. (it serves as the field title rather than being an option within a checkbox group).

<img width="744" height="117" alt="Image" src="https://github.com/user-attachments/assets/56c49dad-1ab0-4bae-b00d-d47af7b1ca4c" />